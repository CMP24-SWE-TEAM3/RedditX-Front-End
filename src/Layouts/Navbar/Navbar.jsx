// imports
import Badge from "Components/BadgeButton/Badge";
import SearchBar from "Components/SearchBar/SearchBar";
import {StyledHeader, StyledSeperator} from "./Navbar.styled";
import PopularButton from "Components/PopularButton/PopularButton"
import CoinButton from "Components/CoinButton/CoinButton";
import ModeratorButton from "Components/ModeratorButton/ModeratorButton";
import CreatePostButton from "Components/CreatePostButton/CreatePostButton"
import ChatButton from "Components/ChatButton/ChatButton";
import NotificationButton from "Components/NotificationButton/NotificationButton";
import AdvertiseButton from "Components/AdvertiseButton/AdvertiseButton";
import DropDownRightButton from "Components/DropDownRight/DropDownRight";
import DropDownLeft from "Components/DropDownButtonLeft/DropDownLeft";
import SignUp from "Components/SignUp/SignUp";
import LoginButton from "Components/LoginButton/LoginButton";
import React, {useContext} from "react";
import AuthContext from "Contexts/auth-context";

/**
 * Component that  Main Links component in header  called navigation bar.
 *
 * @returns {Component.React}
 */
const Navbar = ({theme,toggleMode}) => {
    const ButtonHandler = () => {
        alert('Hello Reddit!');
    }
    const ctx = useContext(AuthContext);
    return (
        <StyledHeader>
            <div className='search'>
                <Badge theme={theme}/>
                <DropDownLeft/>
                <SearchBar/>
            </div>
            {ctx.isLoggedIn &&
                (<>
                    <PopularButton onshow={ButtonHandler}/>
                    <CoinButton onshow={ButtonHandler}/>
                    <StyledSeperator/>
                    <ModeratorButton onshow={ButtonHandler}/>
                    <ChatButton onshow={ButtonHandler}/>
                    <NotificationButton onshow={ButtonHandler}/>
                    <CreatePostButton onshow={ButtonHandler}/>
                    <AdvertiseButton/></>)
            }
            {!ctx.isLoggedIn && <div className={'logged-out'}>
                <SignUp/>
                <LoginButton/>
            </div>}


            <DropDownRightButton toggleMode={toggleMode}/>
        </StyledHeader>
    );
};

export default Navbar;

