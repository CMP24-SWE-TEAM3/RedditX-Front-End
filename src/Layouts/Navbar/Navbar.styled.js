import styled from "styled-components";

export const StyledHeader = styled.header
    `
      justify-content: space-between;
      //position: sticky;
      z-index: 1;
      top: 0;
      left: 0;
      width: 100%;
      height: 49px;
      background-color: ${({theme}) => theme.background.primary};
      color: white;
      display: flex;
      padding: 0 20px;
      align-items: center;
      @media (max-width: 969px) {
        position: fixed;
      }

      .logged-out {
        display: flex;
        justify-content: center;
        align-items: center;
      }

      .search {
        display: flex;
        flex: 1;
      }
    `
export const StyledSeperator = styled.div
    `
      border-left: 1px solid silver;
      height: 75%;
      margin-left: 8px;
      @media(max-width: 990px){
        display: none;
      }
    `