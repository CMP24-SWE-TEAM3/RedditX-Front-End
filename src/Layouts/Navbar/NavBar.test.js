import {shallow} from 'enzyme';

// import Components
import Navbar from "./Navbar";

describe("Navbar Component", () => {
    it("this is a test for Navbar Component", () => {
        expect(shallow(<Navbar/>)).toMatchSnapshot();

    });
});