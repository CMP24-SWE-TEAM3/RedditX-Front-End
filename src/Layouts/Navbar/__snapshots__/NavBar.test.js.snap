// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component this is a test for Navbar Component 1`] = `ShallowWrapper {}`;
